### PR TITLE
Voice Fraud Prevention Guide

### PR DESCRIPTION
**Pull request by vshisterov**
## Description

Basic voice calling fraud prevention hints.

### CODE REVIEW

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+title: Fraud Prevention
+description: Learn basic voice calling fraud prevention practices
+navigation_weight: 10
+---
+
+# Fraud Prevention
+
+Like in every business, there are always persons and companies trying to get profit unfairly in the world of telephony communications. Vonage is performing a set of automatic and manual actions and processes to protect you from fraud activities. Still, there are certain practices to follow which may help you avoid business impact and potential losses.
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:9]: I think we need a better descriptor for the negative practice you are describing. The modifier "unfair" is too vague. Is that an assault on capitalism in general or specific illegal practices?

Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:9]:
```suggestion
Like in every business, there are always persons and companies trying to get profit unfairly in the world of telephony communications. Vonage undertakes a series of automatic and manual actions and processes to protect you from fraud activities. Still, there are certain practices to follow which may help you avoid business impact and potential losses.
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Typical fraud schemes are based on using for outbound calling specific phone numbers, also known as "premium destinations" or "revenue shared numbers", so that the fraudsters get money for each second of the call (with the carrier providing the number being involved or not involved in some cases). To do that, they should either know your API key and secret or use your service UI/API, which allows entering the destination phone number, for example, login or reset password page using two-factor authentication with a voice call. Follow the instruction provided below to reduce the risk of being affected by this scheme.
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:11]: I don't know what this sentence means. Can you please rephrase this part:

> Typical fraud schemes are **based on using for outbound calling specific phone numbers**, also known as "premium destinations" or "revenue shared numbers"

Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:11]: I'm also not sure how to rephrase this sentence but it needs more clarity:

> To do that, they should either know your API key and secret or use your service UI/API, which allows entering the destination phone number, for example, login or reset password page using two-factor authentication with a voice call.

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## Credentials Rotation
+
+If you have any suspicions that your dashboard credentials or API key/secret were compromised, update it as soon as possible. You can change your password in the Dashboard [Edit profile](https://dashboard.nexmo.com/edit-profile) page and request a new API secret on the [Settings](https://dashboard.nexmo.com/settings) page. See also [Best Security Practices for your Vonage Account](https://help.nexmo.com/hc/en-us/articles/115014939548).
+
+## Destinations Filtering
+
+If your business operates in certain countries, you may restrict the destination number to be in these countries. Alternatively, you may restrict certain destination, for example, blocking the following country numbers, which often used for fraud activities:
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:19]:
```suggestion
If your business operates in certain countries, you may restrict the destination number to those countries. Alternatively, you may restrict an entire destination, blocking the following country numbers, which are often used for fraud activities:
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+* Congo
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:21]: General question -- While statistically, this may be true, is it good practice to list these countries as recommended to block entirely? Perhaps some more data references should be added? Otherwise, it does come off as a bit discriminatory.

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+* Ivory Coast
+* Latvia
+* Liberia
+* Samoa
+* the Solomon Islands
+* Vanuatu
+
+> Vonage equally respects all countries and nations; the hint above is based on our statistical data only.
+
+In order to allow/block destinations, you may check the destination number (`from`) before performing Vonage API request for outbound call creation (`POST /calls` request or `connect` NCCO action) either by:
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:34]:
```suggestion
In order to allow/block destinations, you may check the destination number (`from`) before performing a Vonage API request for an outbound call creation (`POST /calls` request or `connect` NCCO action) either by:
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+* first digits in the number - check if it falls under your allowed/blocked destinations,
+* destination country using Number Insights API,
+
+and return an appropriate message if the destination is not supported by your service.
+
+Assuming you want to allow calls to, for example, the United States and Canada only, with the first approach, use this construction:
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:41]:
```suggestion
Assuming you want to allow calls to, for example, the United States and Canada only, with the first approach, you can use this construction:
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md (hunk 4 above)
Comment (vshisterov) [reply _documentation/en/voice/voice-api/guides/fraud-prevention.md:21]: Competitors do it as well, e.g. https://www.twilio.com/docs/usage/anti-fraud-developer-guide#programmable-voice has the list

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md (hunk 1 above)
Comment (vshisterov) [reply _documentation/en/voice/voice-api/guides/fraud-prevention.md]: Well, they definitely are not using the service as it supposed to be used like there is no real communication happening, just silence or fake tones or just random music - so there is no value in these calls in terms of information exchange between people and surely there is no usage of 2FA codes or messages delivered with the call. The only purpose is to get the money from the company(s). Though officially it's not really "illegal" from what I know, still I would say it doesn't fall under "fair usage" - which is a kind of widely used term.

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
+---
+title: Fraud Prevention
+description: Learn basic voice calling fraud prevention practices
```
Comment (marklewin) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:3]:
```suggestion
description: Protect your Voice API applications from fraud
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md (hunk 1 above)
Comment (marklewin) [reply _documentation/en/voice/voice-api/guides/fraud-prevention.md]: If these practices are not really "illegal", are we comfortable calling them "fraudulent"?

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+navigation_weight: 10
+---
+
+# Fraud Prevention
+
+Like in other businesses, there are people and companies trying to profit unfairly in the world of telephony communications. Vonage undertakes a series of automatic and manual actions and processes to protect you from fraud activities. Still, there are certain practices that, if followed, may help you avoid business impact and potential losses.
+
+Typical fraud schemes are based on the generation of outbound calls to specific phone numbers known as "premium destinations" or "revenue shared numbers", so that the fraudsters get money for each second of the call. To do that, fraudsters should either know your API key and secret or use your service UI/API, which allows end-users to enter the destination phone number. Typical services that require this phone number input include a login or reset password page that uses two-factor authentication with a voice call. Follow the instruction provided below to reduce the risk of being affected by these types of schemes.
```
Comment (marklewin) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:11]: Don't say that fraudsters "should" know your API key and secret. They will have gained access to it via fraudulent means.

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+## Credentials Rotation
+
+If you have any suspicions that your dashboard credentials or API key/secret were compromised, update it as soon as possible. You can change your password in the Dashboard [Edit profile](https://dashboard.nexmo.com/edit-profile) page and request a new API secret on the [Settings](https://dashboard.nexmo.com/settings) page. See also [Best Security Practices for your Vonage Account](https://help.nexmo.com/hc/en-us/articles/115014939548).
```
Comment (marklewin) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:15]:
```suggestion
If you have any suspicions that your dashboard credentials or API key/secret have been compromised, update themas soon as possible. You can change your password in the Dashboard [Edit profile](https://dashboard.nexmo.com/edit-profile) page and request a new API secret on the [Settings](https://dashboard.nexmo.com/settings) page. See also [Best Security Practices for your Vonage Account](https://help.nexmo.com/hc/en-us/articles/115014939548).
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+## Destinations Filtering
+
+If your business operates in certain countries, you may restrict the destination number to those countries. Alternatively, you may restrict an entire destination, blocking the following country numbers, which are often used for fraud activities:
```
Comment (marklewin) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:19]:
```suggestion
If your business only operates in certain countries, you might choose to create an "allow list" to restrict calls to those countries. Alternatively, you can create a "block list" that allows calls to be made anywhere except to those countries in the list.

Some users elect to prevent calls from being placed to countries that are perceived as being high risk. These include:
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md (hunk 1 above)
Comment (vshisterov) [reply _documentation/en/voice/voice-api/guides/fraud-prevention.md]: I'm pretty sure we can: 
1) there is no other purpose of doing these calls rather than getting/sharing this premium destination fee: the fraudsters misusing our customer services, e.g. using two-factor authentication calls not to get the temporary password but just to make our customer pay them for the call
2) some of our customers suffer from it for tens and hundreds of thousands euros monthly 
3) our competitors explicitly call it "fraud" as well

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+> Vonage equally respects all countries and nations; the hint above is based on our statistical data only.
+
+In order to allow/block destinations, you may check the destination number (`from`) before performing a Vonage API request for an outbound call creation (`POST /calls` request or `connect` NCCO action) either by:
+
+* first digits in the number - check if it falls under your allowed/blocked destinations,
+* destination country using Number Insights API,
+
+and return an appropriate message if the destination is not supported by your service.
+
+Assuming you want to allow calls to, for example, the United States and Canada only, with the first approach, you can use this construction in JavaScript:
```
Comment (marklewin) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:41]: Refer to allow list and block list rather than first approach/second approach

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+if (phoneNumber.startsWith('1'))
+{
+    vonage.calls.create({ // ...
+    })
+} else {
+    alert("Sorry, the destination is not supported.")
+}
+```
+
+With its comparable simplicity, the drawback here is that, for the example above, `+1` phone country code corresponds not only to the US and Canada but a few other countries/regions which might be outside of your case relevant destinations, for example, the Bahamas or Antigua and Barbuda. To do a more precise check, it's recommended to use basic Number Insights API to check the destination country:
```
Comment (marklewin) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:53]:
```suggestion
The flaw with this approach is that the `+1` phone country code allows calls not only to the US and Canada but also the Bahamas, Antigua and Barbuda, which might not be your intention. You can target countries more precisely by using the Number Insight API:
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+  }
+})
+
+```
+
+See also [Number Insight Basic](/number-insight/code-snippets/number-insight-basic) code snippet.
+
+## Webhook Origination Checking
+
+Other types of attacks might target getting access to your application or customer private data. In this type of attack, a fraudster impersonates Vonage and makes webhook requests to your application similar to Vonage API callbacks, in order to receive valuable information, for example, the text of the voice message you deliver potentially including some Personally Identifiable Information. To make sure the webhook requests to your application are coming from Vonage, you may:
```
Comment (marklewin) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:71]:
```suggestion
Other types of attacks might attempt to gain access to your application or your customers' private data. In these attacks, a fraudster impersonates the Vonage Platform by making webhook requests to your application. These can reveal the voice message text which could contain sensitive data. To ensure that requests to your webhooks are from Vonage, you can:
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+* setup a firewall to accept requests from [Vonage IP ranges](https://help.nexmo.com/hc/en-us/articles/115004859247-Which-IP-addresses-should-I-whitelist-in-order-to-receive-voice-traffic-from-Nexmo-) only;
```
Comment (marklewin) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:73]: Please capitalise first letter in bullet points

**Review by marklewin — CHANGES_REQUESTED** [review]
Some stylistic changes for readability/clarity

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md (hunk 11 above)
Comment (vshisterov) [reply _documentation/en/voice/voice-api/guides/fraud-prevention.md:41]: I meant "check first digits of the number" by "the first approach" here... Anyway seems it needs better wording here, thinking of it now.

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md (hunk 12 above)
Comment (vshisterov) [reply _documentation/en/voice/voice-api/guides/fraud-prevention.md:53]: the Bahamas and Antigua and Barbuda are not the only countries with "+1", just an example (which btw also falls under "risk destination" according to what we see in our customer's experience). Can we keep "for example" somewhere here?

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+> Vonage equally respects all countries and nations; the hint above is based on our statistical data only.
+
+In order to allow/block destinations, you may check the destination number (`from`) before performing a Vonage API request for an outbound call creation (`POST /calls` request or `connect` NCCO action) either by:
+
+* first digits in the number - check if it falls under your allowed/blocked destinations,
+* destination country using Number Insights API,
+
+and return an appropriate message if the destination is not supported by your service.
+
+Assuming you want to allow calls to, for example, the United States and Canada only, with the first approach, you can use this construction in JavaScript:
```
Comment (vshisterov) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:43]: @bencgreenberg could we make anyhow nice tabbed examples in different coding languages here?

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md (hunk 12 above)
Comment (bencgreenberg) [reply _documentation/en/voice/voice-api/guides/fraud-prevention.md]: I think we need to differentiate between a guide to fraud prevention vs a reference doc. I am reading this new file as the former, so it doesn't need to be comprehensive in listing every example in a single reference sentence. That would be better placed in a reference doc, which would read less like prose in any case.

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md (hunk 15 above)
Comment (bencgreenberg) [reply _documentation/en/voice/voice-api/guides/fraud-prevention.md:43]: We could, but see my comment above. I don't think it's necessary for a guide that is functioning as a best practices doc, and not a tutorial or a reference documentation.

Comment (vshisterov) [reply _documentation/en/voice/voice-api/guides/fraud-prevention.md:43]: Makes sense! 👍

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 16]
```diff
@@ -18,20 +18,24 @@ If you have any suspicions that your dashboard credentials or API key/secret hav
...
-* Samoa
-* the Solomon Islands
-* Vanuatu
-
-> Vonage equally respects all countries and nations; the hint above is based on our statistical data only.
+Some users elect to prevent calls from being placed to countries that are perceived as being high risk. 
+
+| ### High Risk Destination Examples
+|
+| These include (but not limited to):
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:25]:
```suggestion
| These include (but are not limited to):
```

File: _documentation/en/voice/voice-api/guides/fraud-prevention.md [hunk 17]
```diff
@@ -0,0 +1,80 @@
+---
+title: Fraud Prevention
+description: Protect your Voice API applications from fraud
+navigation_weight: 10
+---
+
+# Fraud Prevention
+
+Like in other businesses, there are people and companies trying to profit unfairly in the world of telephony communications. Vonage undertakes a series of automatic and manual actions and processes to protect you from fraud activities. Still, there are certain practices that, if followed, may help you avoid business impact and potential losses.
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/fraud-prevention.md:9]: Based on the earlier conversation on the distinction between fraudulent and illegal, can we rephrase this to not be "profit unfairly" as I think that doesn't capture the intention. Profit unfairly could mean they mark up prices too much and customers could save more elsewhere, not fraudulent practices.

**Review by bencgreenberg — APPROVED** [review]
Great job, @vshisterov! This will be a nice contribution to the docs.
